app/app_controller.php

```diff
@@ -4,18 +4,35 @@ class AppController extends Controller
     public $default_view_class = 'AppLayoutView';
 
     function beforeFilter()
-	{
-		$exclude = array(
-			'user/registration',
-			'user/login',
-			'user/delete',
-			'user/confirmation'
-		);
+    {
+        $exclude = array(
+            'user/registration',
+            'user/login',
+            'user/banned'
+        );
 
-		if (in_array(Param::get(DC_ACTION), $exclude)) return;
+        if (in_array(Param::get(DC_ACTION), $exclude)) return;
 
-		if (!isset($_SESSION['userid'])) {
-		    redirect('/user/login');
-		}
-	}
+        if (!isset($_SESSION['userid'])) {
+            redirect('/user/login');
+        } else {
+            $appmodel = new AppModel();
```

Comment:
rather than modifying the `AppModel`, I think it's better to re-use the existing method in User model po :bow:

```php
$user = User::get($user_id);
if ($user->status === ACTIVE) {
```